test/promql/tests/test_controller_reconciliation_errors.yaml

```
@@ -0,0 +1,91 @@
+        values: '1+1x19'
+      - series: 'controller_runtime_reconcile_total{controller="c1-controller", namespace="prod-ns1", pod="pod-1", container="c1", service="s1"}'
+        values: '1+1x19'
+
+    alert_rule_test:
+      - eval_time: 20m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns1
+              pod: pod-1
+              controller: c1-controller
+              container: c1
+              service: s1
+            exp_annotations:
+              summary: "Controller c1-controller has reconciliation errors greater than 1%"
+              description: "Controller c1-controller with Pod pod-1 for namespace prod-ns1 has total reconciliation errors greater than 1% compared to total reconciliations in last 15 minutes."
+
+  # Not Alerted test cases
+  - interval: 1m
+    input_series:
+      # Controller with error throughout with value greater than 1% but it's constant series so it's not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '1x19'
+      - series: 'controller_runtime_reconcile_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '10x19'
+
+      # Controller without error at start and then error starts. Since all 15 mins the errors aren't present, this is not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c2-controller", namespace="prod-ns2", pod="pod-2", container="c1", service="s1"}'
+        values: '0x5 1+1x13'
+      - series: 'controller_runtime_reconcile_total{controller="c2-controller", namespace="prod-ns2", pod="pod-2", container="c1", service="s1"}'
+        values: '1+1x19'
+
+      # Controller without error in the middle for sometime, since all 15 mins the errors aren't present, this is not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c3-controller", namespace="prod-ns3", pod="pod-3", container="c1", service="s1"}'
+        values: '1+1x6 0x5 1+1x6'
+      - series: 'controller_runtime_reconcile_total{controller="c3-controller", namespace="prod-ns3", pod="pod-3", container="c1", service="s1"}'
+        values: '1+1x19'
+
+      # Controller with error throughout but there is a window where the value is less than 1% so this is not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c4-controller", namespace="prod-ns4", pod="pod-4", container="c1", service="s1"}'
+        values: '1+1x19'
+      - series: 'controller_runtime_reconcile_total{controller="c4-controller", namespace="prod-ns4", pod="pod-4", container="c1", service="s1"}'
+        values: '1+1x7 1000x2 1+1x8'
```

If errors are constantly increasing then the total number should increase as well - at least as much as the errors. Should be something like this:
```suggestion
        values: '1+1x7 1000+1x11'
```